z-open/zerv-ng-core
# 5400 - ET -  load balancer distribution

This is  a small lib change to support distributing user session connections (browser tabs) to multiple servers.

Currently Zerv only works with Sticky session at the load balancer level to initiate the socket connection due to a socketio feature (By default, Socketio uses polling then upgrades to web-socket).

<h3 id="steps-to-reproduce">1. Steps to Reproduce</h3>

**Test library**
Make sure the automatic tests pass.
`gulp test`

**Test your application login**
The test cases are in the app pr 5400. 
The objective will be:

- Test with NGINX and AWS ALV  (not tested yet) and multiple application servers. Remove sticky session in ALV or ip_hash  in NGINX
- Check that user can log in and log out
- Log in and open multiple browser tabs, and check that browser tabs are distributed on all servers
- Check that when the user logs out, all tabs are logged out.

<h3 id="performance-considerations">2. Performance Considerations</h3>
With this simple change in the library, long polling will no longer be supported by default.
Browsers must support web socket.

However the function enableLongPolling in the authProvider can enable the previous behavior.

<h3 id="additional-notes">Additional Notes</h3>
